### PR TITLE
core-fixture version 2.5.3 is defined in the build.gradle file, just …

### PR DESCRIPTION
**Pull request by kbayraktar**
…until version 3.0.1 is available under https://jcenter.bintray.com/org/testeditor/fixture/core-fixture/

### CODE REVIEW

**Comment by kbayraktar** [comment]
Unfortunately adding the "new" core-fixture 3.0.1  from bintray to jcenter did not work (yet).

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml [hunk 1]
```diff
@@ -0,0 +1,148 @@
+package org.testeditor.fixture.web
+
+interaction type wait {
+	label = "Wait seconds"
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml]: label is optional. we should remove it

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+component type GeneralBrowser {
+	interactions = wait, goToUrl, closeBrowser, setActiveBrwoser
+}
+
+component type Page {
+	interactions = wait
+}
+
+component WebBrowser is GeneralBrowser {
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml]: We should use the shorter sytax to create an element for a component:
element xyz is type located by locator

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+import org.testeditor.fixture.core.interaction.FixtureMethod;
+
+import io.github.bonigarcia.wdm.ChromeDriverManager;
+import io.github.bonigarcia.wdm.InternetExplorerDriverManager;
+import io.github.bonigarcia.wdm.MarionetteDriverManager;
+
+public class WebDriverFixture {
+
+	private WebDriver driver;
+	private Logger logger = LogManager.getLogger(WebDriverFixture.class);
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: We decided to use slf4j as logging facade

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+		}
+		if (result == null) {
+			result = driver.findElement(By.name(elementLocator));
+		}
+		return result;
+	}
+
+	private void executeScript() {
+		try {
+			System.err.println(exeuteScript);
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: We should replace this with a logger statement

File: src/main/java/org/testeditor/fixture/web/angularjs/NgOneWebDriverFixture.aml [hunk 5]
```diff
@@ -0,0 +1,119 @@
+package org.testeditor.fixture.web.angularjs
+
+interaction type wait {
+	label = "Wait seconds"
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/angularjs/NgOneWebDriverFixture.aml]: We should also remove in this one the label properties

File: src/main/java/org/testeditor/fixture/web/angularjs/NgOneWebDriverFixture.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+	private Logger logger = LogManager.getLogger(NgOneWebDriverFixture.class);
+	private NgWebDriver ngWebDriver;
+
+	@Override
+	protected WebElement getWebElement(String elementLocator) {
+		waitForAngularCompleteOperations();
+		if (elementLocator.startsWith("[model")) {
+			if (elementLocator.startsWith("[model(")) {
+				String substring = elementLocator.substring(elementLocator.indexOf("(") + 1);
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/angularjs/NgOneWebDriverFixture.java]: extracting this to a new method with a clear name, would make it much easier to read this code.

File: src/main/java/org/testeditor/fixture/web/angularjs/materialdesign/NgMdWebDriverFixture.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import org.openqa.selenium.interactions.Actions;
+import org.testeditor.fixture.core.interaction.FixtureMethod;
+import org.testeditor.fixture.web.angularjs.NgOneWebDriverFixture;
+
+import com.paulhammant.ngwebdriver.ByAngular;
+import com.paulhammant.ngwebdriver.ByAngularCssContainingText;
+
+public class NgMdWebDriverFixture extends NgOneWebDriverFixture {
+
+	private Logger logger = LogManager.getLogger(NgMdWebDriverFixture.class);
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/angularjs/materialdesign/NgMdWebDriverFixture.java]: We should use the logging facade

File: src/main/java/org/testeditor/fixture/web/angularjs/materialdesign/NgMdWebDriverFixture.aml [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+	method = NgMdWebDriverFixture.waitSeconds(seconds)
+}
+
+interaction type startBrowser {
+	template = "start browser" ${element} 
+	method = NgMdWebDriverFixture.startBrowser(element)
+}
+
+interaction type startFireForxPortable {
+	label = "Start browser firefox protable"
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/angularjs/materialdesign/NgMdWebDriverFixture.aml]: we should drop out all label statements

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml (hunk 1 above)
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml]: - new discussion point:
I think a label is really helpful when differentiating -> starting with different browser versions.
like this:

	element Firefox is BrowserApplication {
		label = "Firefox(47 and above)"
		locator ="modernfirefox"
	}
as on other hands without a label
like this
	element Firefox is BrowserApplication {
		locator ="modernfirefox"
	}
	element OlderFirefox is BrowserApplication {
		locator ="firefox"
	}

what does OlderFirefox without a label means (every developer, even when he was not involved into these discussions, has to know about the past of the development status. But with a label, he just know, that with this element, just Firefox versions older then  V. 47 are on focus.

I think at this point a label is very helpful.
What do you think ?

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml [hunk 9]
```diff
@@ -0,0 +1,148 @@
+package org.testeditor.fixture.web
+
+interaction type wait {
+	label = "Wait seconds"
+	template = "Wait" ${seconds} "seconds"
+	method = WebDriverFixture.waitSeconds(seconds)
+}
+
+interaction type startBrowser {
+	label = "Start browser"
```
Comment (kpanier) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml]: the other labels ca be removed i think. there is no new infomation then the type or dsl

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml (hunk 2 above)
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml]: New Ticket with [Ticket Number!](https://testeditor.atlassian.net/browse/TE-433)

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml [hunk 10]
```diff
@@ -0,0 +1,136 @@
+package org.testeditor.fixture.web
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml]: We should write a ticket that will correct/adjust the templates according to the conventions we decided on: [http://jira.system.local/browse/OTA-370]

File: src/test/java/org/testeditor/fixture/web/HelperTool.java [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+	}
+
+	private void setProxyValuesForFirefox() {
+		logger.info("setting proxy properties");
+		String PROXY_HOST_HTTP_VALUE = proxyValue;
+		String PROXY_PORT_HTTP_VALUE = proxyPort;
+		String PROXY_HOST_HTTPS_VALUE = proxyValue;
+		String PROXY_PORT_HTTPS_VALUE = proxyPort;
+
+		String PROXY_USER = "http.proxyUser";
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/web/HelperTool.java]: constants?

File: src/test/java/org/testeditor/fixture/web/HelperTool.java [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+package org.testeditor.fixture.web;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HelperTool {
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/web/HelperTool.java]: Name is very generic, how about something like BrowserProperties?

File: src/test/java/org/testeditor/fixture/web/HelperTool.java [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+	private String userName = null;
+	private String passwd = null;
+	private String url = null;
+
+	public void initializeProperties() {
+		this.loadProperties();
+		this.setProxyValuesForFirefox();
+	}
+
+	private void setProxyValuesForFirefox() {
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/web/HelperTool.java]: Name is misleading, how about exportToSystemProperties. If more than the firefox will be adressed, additional parameters can be passed (like BrowserProperties.FirefoxPropKeys or something like that)

File: src/test/java/org/testeditor/fixture/web/HelperTool.java [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+		System.setProperty(PROXY_USER, proxyUserValue);
+		System.setProperty(PROXY_PASSWORD, proxyPasswordValue);
+		System.setProperty(PROXY_HOST_HTTP, PROXY_HOST_HTTP_VALUE);
+		System.setProperty(PROXY_PORT_HTTP, PROXY_PORT_HTTP_VALUE);
+		System.setProperty(PROXY_HOST_HTTPS, PROXY_HOST_HTTPS_VALUE);
+		System.setProperty(PROXY_PORT_HTTPS, PROXY_PORT_HTTPS_VALUE);
+		System.setProperty(PROXY_NONPROXY_HTTP, proxyNonProxyValue);
+	}
+
+	private void loadProperties() {
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/web/HelperTool.java]: how about:
	private void loadProperties() throws IOException {
		Properties prop = new Properties();
		try (InputStream input = getClass().getClassLoader().getResourceAsStream(PROPERTIES_FILE)) {
			prop.load(input);
			setProperties(prop);
		}
	}

File: src/test/java/org/testeditor/fixture/web/VmTest.java [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+	 *    <li>Ngwebdriver 0.9.5</li>
+	 *	  <li>ohne Geckodriver</li>
+	 *  </ul>
+	 *	
+	 * @throws InterruptedException
+	 */
+	//@Test
+	public void firefoxPortableStartAndStopTest() throws InterruptedException {
+
+		String pathFirefoxPortable = "c:\\dev\\tools\\firefox\\FirefoxPortable\\firefox.exe";
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/web/VmTest.java]: this test will probably not run on CI

File: src/test/java/org/testeditor/fixture/web/WebDriverFixtureTest.java [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+	 *	  <li>ohne Geckodriver</li>
+	 *  </ul>
+	 *	
+	 * @throws InterruptedException
+	 */
+	//@Test
+	public void firefoxPortableStartAndStopTest() throws InterruptedException{
+
+		logger.info("Starting Firefox Portable 45.2.0");
+		String pathFirefoxPortable = "c:\\dev\\tools\\firefox\\FirefoxPortable\\firefox.exe";
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/web/WebDriverFixtureTest.java]: This test will probably not run on CI

**Pull request merged**